lib/sampling.h

```diff
@@ -5,17 +5,18 @@
 #include "xorshift.h"
 
 #include <cmath>
+#include <thread>
 #include <tuple>
 
 namespace {
     static constexpr float M_2PI = 2.f * M_PI;
 }
 
+namespace sampling {
 
-class Hemisphere {
-public:
-    // TODO: Docs
-    std::pair<Vec, float> sample() {
+    xorshift64star<float> uniform{4};
```

Comment:
So, I've tested the code. We can use __thread, but we have to declare the constructors of xorshift64star as constexpr. As far as I understand, a __thread variable is initialised at compile time, and then, when a new thread is created, the already initialised variable is just copied.